test/test_qa_utils.py

```diff
@@ -1048,3 +1115,256 @@ def test_mock_keys_not_implemented():
 
     err = MockKeysNotImplemented(keys=['foo', 'bar'], operation="some-operation")
     assert str(err) == 'Mocked some-operation does not implement keywords: foo, bar'
+
+
+class SampleClass:
+
+    def __init__(self, favorite_fruit):
+        self.favorite_fruit = favorite_fruit
+
+    FAVORITE_COLOR = CustomizableProperty('FAVORITE_COLOR', description="the string name of a color")
+
+
+class SampleClass2(SampleClass):
+
+    FAVORITE_SONG = CustomizableProperty('FAVORITE_SONG', description="the string name of a song")
+
+
+class SampleClass3(SampleClass2):
+
+    FAVORITE_COLOR = 'blue'
+    FAVORITE_SONG = 'Jingle Bells'
+
+
+def test_find_field_declared_class():
+
+    thing = SampleClass2(favorite_fruit='orange')
+
+    [kind, value] = UncustomizedInstance._find_field_declared_class(thing, 'favorite_fruit')
+    assert kind == 'instance'
+    assert value == 'orange'
+
+    [kind, value] = UncustomizedInstance._find_field_declared_class(thing, 'FAVORITE_SONG')
+    assert kind == SampleClass2
+    ignored(value)
+
+    [kind, value] = UncustomizedInstance._find_field_declared_class(thing, 'FAVORITE_COLOR')
+    assert kind == SampleClass
+    ignored(value)
+
+    with pytest.raises(RuntimeError):
+        # If we were to search for something that simply wasn't a property, customizable or not,
+        # the search would fail and a RuntimeError would be raised.
+        UncustomizedInstance._find_field_declared_class(thing, 'FAVORITE_SHOW')
+
+
+def test_uncustomized_instance():
+
+    thing = SampleClass2(favorite_fruit=CustomizableProperty('favorite_fruit',
+                                                             description="the string name of a fruit"))
+
+    assert str(UncustomizedInstance(thing, field='FAVORITE_SONG')) == (
+        "Attempt to access field FAVORITE_SONG from class test.test_qa_utils.SampleClass2."
+        " It was expected to be given a custom value in a subclass: the string name of a song."
+    )
```

Comment:
Is `UncustomizedInstance` intended for use in debugging? Can you maybe describe some scenarios where you'd want to use this? I don't feel like I'm getting it from the tests here.

Reply:
It's intended to help you get a decent error message at the right time when you reasonably know that you're only half-implementing something (which is what an abstract class is - a half-implementation). Otherwise you get inscrutable error messages that cost debugging time.